### PR TITLE
Fix RedundantPresenceValidationOnBelongs on two files

### PR DESCRIPTION
**Pull request by cyrillefr**
#### What? Why?
- Contributes to #11482
- First attempt was made by #12380 
-  Continuation of #12428 
- See also #11297 that started the work

#### What should we test?
After migration, all tests should pass and no rubocop offense should be raised.
Run the  rake file, to check it any corrupt data is present.

Note: only 1 migration file here because the null were already enforced at DB level for the `variant_overrides` table.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [X] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

**Comment by cyrillefr** [comment]
Hello @sigmundpetersen , @dacook 

I would a rerun of the checks due to some strange error that do not occur locally :)

**Comment by sigmundpetersen** [comment]
I have re-run the spec a couple of times @cyrillefr but it seems consistent on CI

**Comment by sigmundpetersen** [comment]
Seems there's a fix in #12490, you could wait for that

**Comment by sigmundpetersen** [comment]
Hi @cyrillefr , you can rebase this now to get the fixes from #12490

**Comment by cyrillefr** [comment]
Hello @sigmundpetersen ,

Thank you.

File: db/migrate/20240517121235_require_tax_category_on_tax_rate.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RequireTaxCategoryOnTaxRate < ActiveRecord::Migration[7.0]
+  def change
+    change_column_null :spree_tax_rates, :tax_category_id, false
```
Comment (mkllnk) [inline db/migrate/20240517121235_require_tax_category_on_tax_rate.rb:3]: I double-checked the assumption on all prod servers:
```
sql='select count(*) from spree_tax_rates where tax_category_id is null;'
ansible all_prod -u openfoodnetwork -a "psql -h localhost openfoodnetwork ofn_user -c '$sql'"
```

All clear! :heavy_check_mark:

File: lib/tasks/data/check_missing_required_ids_in_tax_rates.rake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      puts 'Checking for null tax_category_id'
+      ids = Spree::TaxRate.where(tax_category_id: nil).pluck(:id)
+
+      if ids.empty?
+        puts 'No NULL tax_category_id found in spree_tax_rates'
+      else
+        puts 'NULL tax_category_ids s have been found in spree_tax_rates:'
+        print ids
+      end
+    end
```
Comment (mkllnk) [inline lib/tasks/data/check_missing_required_ids_in_tax_rates.rake:16]: Thank your for providing this. But having it in this commit doesn't really help. The rake task can only be run once the code has been deployed. But if you deploy this commit then the migration is run already.

I did a pre-flight check querying all servers via Ansible. Only admins can do that, of course. I'm not sure if there's any better approach.

Can you remove this file again? Or maybe I quickly do it to save time.

**Review by mkllnk — APPROVED** [review]

**Review by dacook — APPROVED** [review]
👌 Perfect

**Comment by dacook** [comment]
Fixing conflict...

File: lib/tasks/data/check_missing_required_ids_in_tax_rates.rake (hunk 2 above)
Comment (cyrillefr) [reply lib/tasks/data/check_missing_required_ids_in_tax_rates.rake]: Hello @mkllnk , I will make these scripts in a separate commits ahead of migrations for next time.
Also, I will make a PR to delete these one-short rake taskes I have created.

**Pull request merged**